webserver.cpp

```diff
@@ -1009,13 +1015,24 @@ int http_parse_multipart_body(struct webserver_t *client, unsigned char *buf, ui
             client->substep = 0;
           } else if(client->ptr == WEBSERVER_BUFFER_SIZE) {
             uint8_t ending = 0;
+	    bool ending3dash = false;
```

Comment:
and then the last changes which are necessary to check for double CRLF or CRLF== markings which can be split due to end of client buffer. Without these checks parts of these markings are read as data and not boundaries.  The \r\n check was already there but the others where necessary also.

Reply:
These additions are indeed nice. I could also easily reproduce it with your additional unittest.